### PR TITLE
Add package for python3-aeneas

### PR DESCRIPTION
**Pull request by StephenMcConnel**
Also fix spurious warning messages that occur for at least python 3.8.

This is needed for the soon arrival of Ubuntu 20.04 which ships with python3 by default.

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,10 @@
+aeneas (1.7.4) stable; urgency=medium
```
Comment (chrisvire) [inline debian/changelog:1]: It should be 1.7.3 since 1.7.4 is in `devel` branch and hasn't been released.

File: debian/control [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
                libasound2-dev, libsndfile1-dev, libespeak-dev
-Standards-Version: 3.9.5
+Standards-Version: 4.1.4
 Homepage: https://github.com/readbeyond/aeneas
 
 Package: python-aeneas
 Architecture: any
 Depends: ${misc:Depends}, ${python:Depends}, 
          espeak, espeak-data, libespeak1, vorbis-tools, ffmpeg, flac
+Conflicts: python3-aeneas
```
Comment (chrisvire) [inline debian/control:21]: Why should they be conflicting? You could have them installed under python2 and python3?

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -1,5 +1,9 @@
 Changelog
 =========
+v1.7.4 (2020-04-02)
```
Comment (chrisvire) [inline docs/source/changelog.rst:3]: This wouldn't be a release of 1.7.4

Comment (StephenMcConnel) [reply docs/source/changelog.rst:3]: There are a couple of trivial code changes from what I started with, so we can't really call it 1.7.3
(which already existed in the changelog, if not in debian packages).  I didn't see that 1.7.4 was taken on a branch somewhere.  If it works, I suppose we could change the version to 1.7.3.1.

File: debian/control (hunk 2 above)
Comment (StephenMcConnel) [reply debian/control:21]: I could be wrong in the details, but I think the shell script in /usr/bin retains the same name regardless of the version of python it calls, but it calls a specific version of python.  My effort was to make the minimal change necessary to get python3-aeneas packages working.
I'm not sure why anyone would need both packages installed simultaneously.
Some of what is going on in python package builds is still magic to me, and I was relieved that it worked as well and as easily as it did even if the resulting packages had a conflict in one or two files.

File: docs/source/changelog.rst (hunk 3 above)
Comment (chrisvire) [reply docs/source/changelog.rst:3]: 1.7.3.1 make sense to me.

**Comment by chrisvire** [comment]
@pettarin, could you accept this PR? We need this change for support Ubuntu Focal 20.04 which doesn't have Python2.

**Comment by pettarin** [comment]
Hi, things are tough here in Italy... I will try to have a look at the 
PR during the weekend.

File: debian/control [hunk 4]
```diff
@@ -1,21 +1,35 @@
...
+Replaces: python3-aeneas
+Description: Python library to automagically synchronize audio and text
+ aeneas automatically generates a synchronization map between a list of
+ text fragments and an audio file containing the narration of the text.
+ In computer science this task is known as (automatically computing a)
+ forced alignment.
+
+Package: python3-aeneas
+Architecture: any
+Depends: ${misc:Depends}, ${python3:Depends},
```
Comment (chrisvire) [inline debian/control:30]: Needs `${shlibs:Depends}` to work around lintian error: `missing-dependency-on-libc`

File: debian/control [hunk 5]
```diff
@@ -1,21 +1,35 @@
...
+               python3-setuptools,
+               python3-numpy, python3-lxml, python3-bs4,
                libasound2-dev, libsndfile1-dev, libespeak-dev
-Standards-Version: 3.9.5
+Standards-Version: 4.1.4
 Homepage: https://github.com/readbeyond/aeneas
 
 Package: python-aeneas
 Architecture: any
 Depends: ${misc:Depends}, ${python:Depends},
```
Comment (chrisvire) [inline debian/control:19]: Needs `${shlibs:Depends}` to work around lintian error: `missing-dependency-on-libc`

File: debian/rules [hunk 6]
```diff
@@ -3,5 +3,5 @@
 export PYBUILD_NAME=aeneas
 
 %: 
-	dh $@ --with python2 --buildsystem=pybuild
+	dh $@ --with python2,python3 --buildsystem=pybuild
```
Comment (chrisvire) [inline debian/rules:7]: Need one of these to resolve lintian error: `missing-dependency-on-numpy-abi`
```
override_dh_shlibdeps:
        dh_shlibdeps && dh_numpy && dh_numpy3
```
or
```
override_dh_python2:
	dh_python2
	dh_numpy

override_dh_python3:
	dh_python3
	dh_numpy3
```

**Review by chrisvire — CHANGES_REQUESTED** [review]
Since the rules is creating two packages, the `install` script needs to be changes:

1. copy debian/install to debian/python3-aeneas.install
2. rename debian/install to debian/python-aeneas.install
3. in debian/python3-aeneas.install, change all python-aeneas to python3-aeneas.

This way the `/usr/share/python-aeneas` and /usr/share/phython3-aeneas` folders are installed correctly for the python-aeneas and python3-aeneas packages respectively.

File: debian/control (hunk 4 above)
Comment (StephenMcConnel) [reply debian/control]: added.

File: debian/control (hunk 5 above)
Comment (StephenMcConnel) [reply debian/control]: done.

File: debian/rules (hunk 6 above)
Comment (StephenMcConnel) [reply debian/rules:7]: added.

File: debian/control (hunk 2 above)
Comment (StephenMcConnel) [reply debian/control]: changed Conflicts: to Replaces: for both binary packages

File: debian/changelog [hunk 7]
```diff
@@ -1,3 +1,10 @@
+aeneas (1.7.3.1) stable; urgency=medium
+
+  * Fix spurious warnings about "encoding is not UTF-8" when encoding is "utf-8"
+  * Add debian packaging for python3-aeneas
+
+ -- Stephen McConnel <stephen_mcconnel@sil.org>  Thu, 02 Apr 2020 16:45:00 -0600
+
```
Comment (chrisvire) [inline debian/changelog:7]: @StephenMcConnel, Drat, I changed this while testing and forgot to include here:

`E: python3-aeneas: debian-changelog-file-contains-invalid-email-address alberto@nievo`

So `alberto@nievo` should be changed to `alberto@readbeyond.it` 

I got that address from https://www.albertopettarin.it/contact.html

**Review by chrisvire — CHANGES_REQUESTED** [review]
Sorry, one more change.  I had changed it locally and forgot to document it.

**Comment by pettarin** [comment]
@chrisvire @StephenMcConnel Hi, thanks a lot for taking care of the necessary changes. I will going to merge this PR as soon as you confirm it is ready on your side. ``alberto@readbeyond.it`` is good indeed.

Note that tomorrow (Sunday 2020-05-10) I will also update a bit the documentation, and include new languages, since eSpeak-ng now supports a lot more: https://github.com/espeak-ng/espeak-ng/blob/master/docs/languages.md

If I manage, I will try to release on PyPI by the end of next week --- say, Sunday 2020-05-17.

**Comment by pettarin** [comment]
@chrisvire @StephenMcConnel  also, might I ask you to open the PR against the ``devel`` branch, i.o. directly ``master``, please?

**Comment by chrisvire** [comment]
@pettarin: we can open another PR agains `devel` as well. We will likely want to have a published package using 1.7.3 and then have some time to test against 1.7.4.

Thanks,
Chris

File: debian/changelog (hunk 7 above)
Comment (StephenMcConnel) [reply debian/changelog:7]: I added the line to the lintian-overrides files.

**Comment by chrisvire** [comment]
@pettarin: This looks all good to me.  Can you merge it?

**Comment by readbeyond** [comment]
@chrisvire @StephenMcConnel  Merged, thank you! I will credit you in the documentation, as of 1.7.4. (I planned to work towards 1.7.4 last Sunday, but I had to deal with an emergency and my plans went south. I hope the upcoming weekend will be easier on me.) I understand that you do not need a PyPI release, for the deb package the merge on master was sufficient, isn't it?

**Pull request merged**